ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/EvaluatorImpl.java

```diff
@@ -83,6 +83,9 @@ public String evaluate(String source) throws BallerinaShellException {
     public ShellCompilation getCompilation(String source) {
         PackageCompilation compilation;
         ExceptionStatus exceptionStatus;
+
+        // need to do this first to address fails of tree parser and snippet factoryee
+        invoker.clearPreviousVariablesAndModuleDclnsNames();
```

Comment:
Minor suggestion, can't we add this line inside the `getValueAsObject` method after we call `invoker.resetDiagnostics()` ?